src/mscorlib/shared/System/IO/BinaryWriter.cs

```diff
@@ -389,6 +390,62 @@ public unsafe virtual void Write(String value)
             }
         }
 
+        public virtual void Write(ReadOnlySpan<byte> value)
+        {
+            if (GetType() == typeof(BinaryWriter))
+            {
+                OutStream.Write(value);
+            }
+            else
+            {
+                int length = value.Length;
+                byte[] buffer = ArrayPool<byte>.Shared.Rent(length);
+                try
+                {
+                    value.CopyTo(buffer);
+                    Write(buffer, 0, length);
+                }
+                finally
+                {
+                    ArrayPool<byte>.Shared.Return(buffer);
+                }
+            }
+        }
```

Comment:
> so I think it's faster if the value is assigned to a local

That's not always the case: in fact, sometimes that defeats optimizations of the JIT, in particular with well-known types like strings, arrays, and spans.